lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java

```diff
@@ -561,7 +561,7 @@ protected Schema createSchema(Type type, Map<String,Schema> names) {
         return Schema.create(Schema.Type.BYTES);
       if (Collection.class.isAssignableFrom(c))              // array
         throw new AvroRuntimeException("Can't find element type of Collection");
-      for (Conversion<?> conversion : conversions.values()) {  // logical type
+      for (Conversion<?> conversion : getAllConversions()) { // logical type
```

Comment:
Yes, and this is actually a bug. Originally, the `getConversionTo` and `getConversionFrom` methods implemented the correct `isAssignableFrom` check to work with subclasses. But that's actually the wrong thing to do and we should have a conversion for each class, which is what I changed to. That's both faster and more correct (consider the case where I have PointXY and a subclass PointXYZ: supporting a subclass would discard data).